<?xml version="1.0"?>
<launch>
  <node pkg="rosserial_arduino" type="serial_node.py" name="motor_arduino" output="screen">
    <param name="port" type="string" value="/dev/ttyUSB0" />
    <param name="min_r" type="int" value="$(arg min_r)"/>
    <param name="max_r" type="int" value="$(arg max_r)"/>
  </node>
  
  <node pkg="project_2dto3d" type="change_finder" name="change_finder" output="screen">
    <param name="serial_port_name" type="string" value="/dev/ttyUSB1"/>
    <param name="pub_topicname" type="string" value="/pcl_rviz"/>
    <param name="min_r" type="int" value="$(arg min_r)"/>
    <param name="max_r" type="int" value="$(arg max_r)"/>
    <param name="height" type="double" value="$(arg height)"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find project_2dto3d)/rviz/change_config.rviz"  />

  <node pkg="rosbag" type="play" name="play" output="screen" args="/home/aa/$(arg bag_filename).bag --wait-for-subscribers" />

</launch>
